ui/src/app/settings/containers/Users/Users.js

```diff
@@ -14,54 +15,85 @@ import Pagination from "app/base/components/Pagination";
 import MainTable from "app/base/components/MainTable";
 import Row from "app/base/components/Row";
 
-const generateUserRows = (users, authUser) =>
-  users.map(user => ({
-    columns: [
-      {
-        content: <Link to={`/users/${user.id}`}>{user.username}</Link>,
-        role: "rowheader"
-      },
-      { content: user.email },
-      { content: 37, className: "u-align--right" },
-      { content: "Local" },
-      { content: "12 mins ago" },
-      { content: user.is_superuser ? "Admin" : null },
-      { content: user.sshkeys_count, className: "u-align--right" },
-      {
-        content: (
-          <>
-            <Link
-              to={`/users/${user.id}/edit`}
-              className="p-button--base is-small"
-            >
-              <i className="p-icon--edit">Edit</i>
-            </Link>
-            {user.id !== authUser.id && (
-              <Link
-                to={`/users/${user.id}/delete`}
-                className="p-button--base is-small"
+const generateUserRows = (users, authUser, expandedId, setExpandedId) =>
+  users.map(user => {
+    const expanded = expandedId === user.id;
+    return {
+      className: expanded ? "user-list__expanded-row" : null,
+      columns: [
+        {
+          content: <Link to={`/users/${user.id}`}>{user.username}</Link>,
+          role: "rowheader"
+        },
+        { content: user.email },
+        { content: 37, className: "u-align--right user-list__cell--small" },
+        { content: "Local", className: "user-list__cell--small" },
+        { content: "12 mins ago" },
+        {
+          content: user.is_superuser ? "Admin" : null,
+          className: "user-list__cell--small"
+        },
+        {
+          content: user.sshkeys_count,
+          className: "u-align--right user-list__cell--small"
+        },
+        {
+          content: (
+            <>
+              <Button
+                appearance="base"
+                element={Link}
+                to={`/users/${user.id}/edit`}
+                className="is-small user-list__button"
               >
-                <i className="p-icon--delete">Delete</i>
-              </Link>
-            )}
-          </>
-        ),
-        className: "u-align--right"
+                <i className="p-icon--edit">Edit</i>
+              </Button>
+              {user.id !== authUser.id && (
+                <Button
+                  appearance="base"
+                  className="is-small user-list__button"
+                  onClick={() => setExpandedId(user.id)}
```

Comment:
Your giving each row a lot of power here. They shouldn't have the ability to set other row. Maybe think about moving this state to the table component itself? And then just passing a click function to the rows?

Reply:
There are a couple of issues there, one is that we don't have a consistent element that opens the row (in this case it is the trash icon).

The second is that we need to be able to control that state outside the table. For this list when the user confirms the delete then we need to close that row.